scripts/src/main/resources/scripts/command/docker

```diff
@@ -17,84 +17,107 @@ source "$(dirname "${0}")"/../functions
 # $1: optional setup
 function doSetup() {  
   if doIsDockerInstalled 
-  then     
-    doEcho "docker is already installed at $(command -v docker)" 
-    exit 0
+  then
+    doEcho "docker is already installed at $(command -v docker)"
   else
-    if doIsBatch
-    then
-      doFail "Interactive installation is required, cannot proceed in batch mode. Please rerun without batch option."
-    fi
     #Start setup
     if doIsWindows
-    then 
-      if doIsWSLEnabled
+    then
+      doRequireWsl
+      # Verify if docker is installed on Windows Subsystem for Linux(WSL)
+      docker_win_version=$(wsl docker version 2>/dev/null)
+      if [[ $docker_win_version == *"Client:"* ]]
+      then
+        if [ "${1}" = "setup" ]
+        then
+          doEcho "Docker already installed on WSL."
+          wsl docker version
+        fi
+        if [[ $docker_win_version != *"Server:"* ]]
+        then
+          doEcho "Please start Docker engine on WSL."
+        fi
+      else
+        doRequireNotBatch
+        doEcho "Installing Docker on WSL..."
+        # Install Docker on WSL
+        doRunCommand "wsl bash ../scripts/setup-docker"
+      fi
+    elif doIsMacOs
+    then
+      doRequireNotBatch
+      local downloadUrl
+      local arch
+      arch="$(uname -m)"
+      if [ "${arch}" = "x86_64" ]
+      then
+        downloadUrl="https://desktop.docker.com/mac/stable/amd64/Docker.dmg?utm_source=docker&utm_medium=webreferral&utm_campaign=docs-driven-download-mac-amd64"
```

Comment:
I think you can just us https://desktop.docker.com/mac/stable/amd64/Docker.dmg
The parameters are just google analytics tracking parameters

Reply:
Excellent feedback. 👍 GET parameters removed.